File: katas/NextBiggerNumber/solutions/marartz_fsharp/NextBiggerNumberTest/NextBiggerNumberTests.fs
module NextBiggerNumberTests

open Xunit

open NextBiggerNumber

[<Fact>]
let ``12 => 21`` () =
    Assert.True((getNextBiggerNumber 12L) = 21L)

[<Fact>]
let ``513 => 531`` () =
    Assert.True((getNextBiggerNumber 513L) = 531L)

[<Fact>]
let ``2017 => 2071`` () =
    Assert.True((getNextBiggerNumber 2017L) = 2071L)

[<Fact>]
let ``459853 => 21`` () =
    Assert.True((getNextBiggerNumber 459853L) = 483559L)

[<Fact>]
let ``59884848459853 => 59884848483559`` () =
    Assert.True((getNextBiggerNumber 59884848459853L) = 59884848483559L)

[<Fact>]
let ``9 => -1`` () =
    Assert.True((getNextBiggerNumber 9L) = -1L)

[<Fact>]
let ``111 => -1`` () =
    Assert.True((getNextBiggerNumber 111L) = -1L)

[<Fact>]
let ``531 => -1`` () =
    Assert.True((getNextBiggerNumber 531L) = -1L)

File: katas/NextBiggerNumber/solutions/markushl/NextBiggerNumber/NextBiggerNumber.Tests/NextBiggerNumber.Tests.cs
namespace NextBiggerNumber.Tests
{
    using Xunit;
    using NextBiggerNumber;

    public class UnitTestNextBiggerNumber
    {


        [Theory]
        [InlineData(12L, 21L)]
        [InlineData(513L, 531L)]
        [InlineData(2017L, 2071L)]
        [InlineData(459853L, 483559L)]
        [InlineData(59884848459853L, 59884848483559L)]
        [InlineData(9L, -1L)]
        [InlineData(11L, -1L)]
        [InlineData(531L, -1L)]
        public void ShouldGetNextBiggerNumber(long nNumber, long nExpected)
        {
            Assert.Equal(Numbers.GetNextBiggerNumber(nNumber), nExpected);
        }

    }
}
